Article/qd je serai grand (#10)
philippart-s/blog: _posts/2020-08-10-Qd-je-serai-grand.md

Ca dure 5 ans, puis, n'étant pas un parisien de souche, l'envie de revenir dans la région étudiante et plus particulièrement la belle ville de [Tours](https://www.tours.fr/){:target="_blank"} se fait de plus en plus sentir. Je saute le pas et reviens donc à Tours dans une société d'édition de logiciel comme expert Java / J2E (on disait comme ça à l'époque :wink:).

Si vous avez bien suivi j'ai 30 ans et le dev est ma principale activité mais avec l'âge et l'expérience on commence à me demander plus de choses et plus de responsabilités. Pas de problème, c'est le jeu et si en plus cela permet de distiller de la connaissance que j'ai acquise on est bien dans les principes qui m'animent.

### Le début de la fin
Eh oui on se doute qu'il y a bien un truc qui va changer, sinon pourquoi faire un article ? :wink:

> Avec l'âge et l'expérience qui augmentent vient la question de la juste rémunération, n'étant plus un développeur junior forcément je pense (et j'espère que c'est vrai) avoir acquis un bon niveau d'expertise dans le développement (Java pour ma part) et donc, comme souvent, les demandes d'augmentations sont liées à une augmentation de responsabilités. Cela ne me pose pas problème : ok !

Reviewer: ```suggestion
Avec l'âge et l'expérience qui augmentent, vient la question de la juste rémunération. N'étant plus un développeur junior, je pense (et j'espère que c'est vrai) avoir acquis un bon niveau d'expertise dans le développement (Java pour ma part). Comme souvent, les demandes d'augmentations sont liées à une augmentation de responsabilités. Cela ne me pose pas problème : ok !
```

Prenons un peu de temps sur cette étape: est-ce vraiment normal que le salaire soit conditionné aux responsabilités ? Ne peut il pas aussi tout simplement l'être à la compétence ? 

> Je vais faire l'analogie avec un artisan : n'est-ce pas normal de rétribuer un ébéniste qui va me faire la même pièce mais plus vite, de meilleure qualité et plus robuste dans le temps ? Oui cela me semble évident que pour un employeur il est normal que cet ébéniste soit payé plus cher qu'un ébéniste qui va travailler moins bien ou moins vite. Mais pour autant si mon ébéniste qui découvrait le métier au bout de quelques années est devenu très bon va-t-on lui demander en plus de gérer les relations fournisseurs, client, la compta, ... pour justifier une augmentation de son salaire, je ne pense pas !

Reviewer: ```suggestion
Je vais faire l'analogie avec un artisan : n'est-ce pas normal de rétribuer un ébéniste qui va me faire la même pièce mais plus vite, de meilleure qualité et plus robuste dans le temps ? Oui cela me semble évident que pour un employeur il est normal que cet ébéniste soit payé plus cher qu'un ébéniste qui va travailler moins bien ou moins vite. Mais pour autant, si mon ébéniste est devenu très bon au fil des années, va-t-on lui demander en plus de gérer les relations fournisseurs, client, la compta, ... pour justifier une augmentation de son salaire? Je ne pense pas !
```

Cependant il n'est pas choquant de payer plus cher cet ébéniste, qui, mécaniquement, va rapporter plus grâce à son expertise et ses compétences.

> A l'époque ou le *plus de responsabilités* rimait avec gérer une équipe, faire des specs, des réunions clients, des chiffrages financiers, ... j'aurai dû sentir le truc venir mais bon j'avais encore du temps pour développer et j'avais cette étiquette de référent technique ou tech lead comme disent les jeunes (ça fait toujours plaisir à l'ego :wink:) qui permettait de me faire sentir encore développeur.

Reviewer: ```suggestion
A l'époque où le *plus de responsabilités* rimait avec gérer une équipe, faire des specs, des réunions clients, des chiffrages financiers, ... j'aurai dû sentir le truc venir ! Mais j'avais encore du temps pour développer et j'avais cette étiquette de référent technique ou tech lead - comme disent les jeunes - (ça fait toujours plaisir à l'ego :wink:) qui permettait de me faire sentir encore développeur.
```

### Le changement ... PowerPoint à la place d'Intelliji

> Là où ça commence à se gâter c'est quand vous vous rendez compte à la fin de la journée que vous n'avez pas lancé votre éditeur de code préféré mais simplement la suite office et plus particulièrement PowerPoint !

Reviewer: ```suggestion
Tout cela commence à se gâter, lorsque vous vous rendez compte qu'à la fin de la journée vous n'avez pas lancé votre éditeur de code préféré mais simplement la suite office et plus particulièrement PowerPoint !
```

> Ca y est j'ai de l'expérience (plus de 15 ans) et d'évolutions et en évolutions je suis devenu architecte (applicatif, logiciel, technique, choisissez celui qui vous plaît le plus :wink:). Je ne rentrerai pas dans le détail du poste d'architecte qui varie selon l'entreprise dans laquelle on se trouve mais j'ai surtout l'impression qu'une fois de plus c'est pour flatter un peu l'ego que ce poste existe :smiley:.

Reviewer: ```suggestion
J'ai maintenant plus de 15 ans d'expérience et suis devenu architecte (applicatif, logiciel, technique, choisissez celui qui vous plaît le plus :wink:). Je ne rentrerai pas dans le détail du poste d'architecte, qui varie selon l'entreprise dans laquelle on se trouve, mais j'ai surtout l'impression, qu'une fois de plus, c'est pour flatter un peu l'ego que ce poste existe :smiley:.
```

### Le développeur, sportif de haut niveau ?

> J'ai un peu l'impression d'être un sportif de haut niveau, non pas dans les performances (:sweat_smile:) mais dans le fait de devoir arrêter ma passion juste parce-que je suis trop vieux. Et du coup pour rester dans le milieu j'accepte des postes périphériques à ma passion d'origine (un peu comme coach, agent, préparateur pour les sportifs).

Reviewer: ```suggestion
J'ai un peu l'impression d'être un sportif de haut niveau, non pas dans les performances (:sweat_smile:) mais dans le fait de devoir arrêter ma passion juste parce-que je suis trop vieux. Et du coup pour rester dans le milieu, j'accepte des postes périphériques à ma passion d'origine (un peu comme coach, agent, préparateur pour les sportifs).
```

> Ce qui est dommage c'est que là où pour un sportif cela peut se justifier par les performances brutes par rapport à l'âge, pour un développeur je ne pense pas que cela soit le cas. Certes les jeunes poussent mais j'estime être encore capable de développer et de manière qualitative.

Reviewer: ```suggestion
Ce qui est dommage, c'est que là où pour un sportif cela peut se justifier par les performances brutes par rapport à l'âge, pour un développeur je ne pense pas que cela soit le cas. Certes les jeunes poussent mais j'estime être encore capable de développer et de manière qualitative.
```

### Redevenir sportif amateur

> Pour continuer dans l'analogie du sportif j'ai l'impression que la seule façon de coder de nouveau est de le faire "en amateur" sur mon temps perso. Le problème est que la vie perso quand on vieilli est occupée (famille, amis, ...) et qu'il est de plus en plus difficile de trouver du temps (j'avoue aussi la motivation, je suis assez mauvais pour les "hello world", je préfère les vrais projets !).

Reviewer: ```suggestion
Pour continuer dans l'analogie du sportif, j'ai l'impression que la seule façon de coder de nouveau est de le faire "en amateur" sur mon temps personnel. Le problème est que la vie perso, quand on vieillit, est très occupée (famille, amis, ...) et qu'il est de plus en plus difficile de trouver du temps (j'avoue aussi la motivation, je suis assez mauvais pour les "hello world", je préfère les vrais projets !).
```

### En conclusion : est-ce normal ?

> Je ne pense pas que cela soit normal dans notre métier, si les compétences sont toujours là, de devoir ne plus coder à cause de l'âge ou des prétentions salariales (bien entendu il faut savoir rester raisonnable :wink:).

Reviewer: ```suggestion
Je ne pense pas que cela soit normal, dans notre métier, de devoir ne plus coder à cause de l'âge ou des prétentions salariales si les compétences sont toujours là (bien entendu il faut savoir rester raisonnable :wink:).
```

> Lorsque l'on me demande ce que je suis naturellement je réponds *développeur* puis je me ravise en disant architecte (en ayant du mal à définir réellement à quoi cela correspond !).

Reviewer: ```suggestion
Lorsque l'on me demande ce que je suis, je réponds naturellement *développeur* puis je me ravise en disant architecte (en ayant du mal à définir réellement à quoi cela correspond !).
```

> Je pense qu'il est possible de continuer à développer, même passé 20 ans d'expérience, et de continuer à apporter quelque chose. J'aime le développement, le *clean code* qui apporte quelque chose, les tests unitaires, ... Je pense, qu'en ces jours où le développement par *StackOverFlow* se rencontre de plus en plus, que comprendre ce que l'on fait et le faire par envie et passion est un réel plus.

Reviewer: ```suggestion
Je pense qu'il est possible de continuer à développer, même passé 20 ans d'expérience, et de continuer à apporter quelque chose. J'aime le développement, le *clean code* qui apporte quelque chose, les tests unitaires, ... Je pense, qu'en ces jours où le développement par *StackOverFlow* se rencontre de plus en plus, comprendre ce que l'on fait et le faire par envie et passion est un réel plus.
```

Cela va certainement faire *vieux con* mais en ces périodes où un nouveau langage sort tous les 3 mois et devient le super langage qu'il faut absolument utiliser, un peu d'expérience sur des langages qui ont fait leurs preuves ne fait pas de mal.

> Il y a forcément pleins de contre exemples mais quand je fais le compte autour de moi de mes amis de promo ou de collègues de mon âge peu sont encore développeurs. Peut être aussi la localisation explique des choses et que ce phénomène se voit moins sur Paris ou les grandes villes, je n'ai pas de chiffres mais là encore par rapport à mes constations sur les différents salons (à l'époque où on pouvait y aller :wink:) j'ai l'impression que les quarantenaires ne sont pas les plus nombreux à être 100 % développeurs !

Reviewer: ```suggestion
Il y a forcément beaucoup de contre-exemples mais quand je fais le compte autour de moi, de mes amis de promo ou de collègues de mon âge, peu sont encore développeurs. La localisation de nos postes peut aussi expliquer des choses et il est possible que ce phénomène se voie moins sur Paris ou les grandes villes, je n'ai pas de chiffres. Mais là encore, par rapport à mes constations sur les différents salons (à l'époque où on pouvait y aller :wink:), j'ai l'impression que les quarantenaires ne sont pas les plus nombreux à être 100 % développeurs !
```

> Alors oui il faut que les jeunes diplômés développent, nous pousse à continuer à faire de la veille et de la R&D pour ne pas s'endormir sur nos lauriers. Il faut aussi du monde pour accompagner ces jeunes et ne pas les laisser seuls avec pour seul compagnon *StackOverFlow*.

Reviewer: ```suggestion
Alors oui : il faut que les jeunes diplômés développent, nous poussent à continuer à faire de la veille et de la R&D pour ne pas s'endormir sur nos lauriers. Il faut aussi du monde pour accompagner ces jeunes et ne pas les laisser seuls avec pour seul compagnon *StackOverFlow*.
```
